OONIProbeUnitTests/APITest.m

```diff
@@ -0,0 +1,49 @@
+#import <XCTest/XCTest.h>
+#import "Measurement.h"
+#import "TestUtility.h"
+#define EXISTING_REPORT_ID @"20190113T202156Z_AS327931_CgoC3KbgM6zKajvIIt1AxxybJ1HbjwwWJjsJnlxy9rpcGY54VH"
+#define NONEXISTING_REPORT_ID @"EMPTY"
+
+@interface APITest : XCTestCase
+
+@end
+
+@implementation APITest
+
+- (void)setUp {
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+}
+
+- (void)testExisting {
+    Measurement *measurement = [Measurement new];
+    measurement.report_id = EXISTING_REPORT_ID;
+    [measurement getExplorerUrl:^(NSString *measurement_url){
+        XCTAssert(true);
+    } onError:^(NSError *error){
+        XCTAssert(false);
+    }];
+}
+
+- (void)testNonExisting {
+    Measurement *measurement = [Measurement new];
+    measurement.report_id = NONEXISTING_REPORT_ID;
+    [measurement getExplorerUrl:^(NSString *measurement_url){
+        XCTAssert(false);
+    } onError:^(NSError *error){
+        XCTAssert(true);
+    }];
+}
+
+-(void)testDownloadUrls{
+    [TestUtility downloadUrls:^(NSArray *urls) {
```

Comment:
Please, create different test classes. Create one for `Measurement` and one for `TestUtility`.

Reply:
These are API tests, they belong together. The function in `Measurement` could have been written in the TestUtility class (like I did in the first commit) or vice versa. Eventually the `TestUtility` class has to be refactored

Reply:
I really do not understand all this resistance to making things right. Also:

> X is fine [...] because eventually will be refactored

is the biggest lie in the history of software engineering. Why not trying to do things right _now_?

Reply:
What I'm saying is that it makes sense to keep all the API related calls in one class rather than around many classes. And in this case we'll need just one test.

Reply:
> it makes sense to keep all the API related calls in one class

Why? Is this an iOS pattern I'm not aware of?

My instinct here suggests rather the opposite. When I am looking for tests for `.../X.m` I usually search for the corresponding `...Tests/XTest.m` file.

Reply:
Well it's a nice patterns, as the `TestUtility` class was meant to be for something else. 
And was not coded at all in Android as xan disliked that pattern at all.
Now it's collecting a lot of functions "I don't know where this belongs"

Reply:
I understand. Can you add a `// TODO(lorenzoPrimi): ...` comment indicating that we should perform this kind of refactor, so that a reader stumbling upon that is not surprised?